--- recode-3.5d/configure.in.orig	Wed Jun 28 16:36:30 2000
+++ recode-3.5d/configure.in	Mon Jul 29 11:45:34 2002
@@ -32,7 +32,6 @@
 
 jm_FUNC_MALLOC
 jm_FUNC_REALLOC
-AC_SUBST(LIBOBJS)
 
 fp_OS_MICROSOFT
 AC_MSG_CHECKING(the default charset)
@@ -52,12 +51,12 @@
 
 # This is necessary so that .o files in LIBOBJS are also built via
 # the ANSI2KNR-filtering rules.
-LIBOBJS=`echo $LIBOBJS | sed 's/\.o /\$U.o /g;s/\.o$/\$U.o/'`
+LIB@&t@OBJS=`echo $LIB@&t@OBJS | sed 's/\.o /\$U.o /g;s/\.o$/\$U.o/'`
 
 LTALLOCA=`echo $ALLOCA | sed 's/\.o/.lo/g'`
 AC_SUBST(LTALLOCA)
-LTLIBOBJS=`echo $LIBOBJS | sed 's/\.o /.lo /g;s/\.o$/.lo/'`
-AC_SUBST(LTLIBOBJS)
+LTLIB@&t@OBJS=`echo $LIB@&t@OBJS | sed 's/\.o /.lo /g;s/\.o$/.lo/'`
+AC_SUBST(LTLIB@&t@OBJS)
 
 AC_OUTPUT(Makefile contrib/Makefile doc/Makefile i18n/Makefile lib/Makefile
 libiconv/Makefile m4/Makefile src/Makefile tests/Makefile tests/atconfig)
--- recode-3.5d/acinclude.m4.orig	Mon Jul 29 11:42:18 2002
+++ recode-3.5d/acinclude.m4	Mon Jul 29 11:47:01 2002
@@ -120,8 +120,7 @@
 	 jm_cv_func_working_malloc=no)
   ])
   if test $jm_cv_func_working_malloc = no; then
-    AC_SUBST(LIBOBJS)
-    LIBOBJS="$LIBOBJS malloc.$ac_objext"
+    AC_LIBOBJ([malloc])
     AC_DEFINE_UNQUOTED(malloc, rpl_malloc,
       [Define to rpl_malloc if the replacement function should be used.])
   fi
@@ -156,8 +155,7 @@
 	 jm_cv_func_working_realloc=no)
   ])
   if test $jm_cv_func_working_realloc = no; then
-    AC_SUBST(LIBOBJS)
-    LIBOBJS="$LIBOBJS realloc.$ac_objext"
+    AC_LIBOBJ([realloc])
     AC_DEFINE_UNQUOTED(realloc, rpl_realloc,
       [Define to rpl_realloc if the replacement function should be used.])
   fi
@@ -255,7 +253,7 @@
       with_included_gettext=yes
     fi
     if test $with_included_gettext = yes; then
-      LIBOBJS="$LIBOBJS gettext.o"
+      AC_LIBOBJ([gettext])
       AC_DEFINE(HAVE_GETTEXT)
       AC_DEFINE(HAVE_DCGETTEXT)
     else
